src/disk_cache.rs

```diff
@@ -0,0 +1,1015 @@
+use bytes::Bytes;
```

Comment:
i think this file is better to be renamed as `cacheable_object_store.rs` but i'm afraid whether this renaming will affect the review history or not.

i'd like to make this rename after the current review comments get resolved.